### PR TITLE
Streamfield provider

### PR DESCRIPTION
**Pull request by cadecairos**
Summary:

This PR adds a custom Faker provider that can output streamfield JSON. As I write this, it currently supports the following components: header, paragraph, image, spacer, image_text2, quote.

Right now the CMSPageFactory class is the only factory using this provider, but we can use it anywhere we want. It's also very easy to add more components as we develop them.

There's an additional step in the load_fake_data command that generates 20 images with random colors, this is done so that a variety of images can be associated with streamfield image components.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 language: python
 python:
 - '3.6'
-services:
-- postgresql
+addons:
+  postgresql: "9.6"
```
Comment (patjouk) [inline .travis.yml]: Why the change to a specific version of postgresql?

File: .travis.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ install:
 - pipenv run python network-api/manage.py collectstatic --no-input
 before_script:
 - psql -c 'create database network;' -U postgres
+- pipenv run python network-api/manage.py migrate --no-input
```
Comment (patjouk) [inline .travis.yml]: It's not necessary: migrations will be applied when the test database is created (run it with verbose level 2 if you're curious)

Comment (cadecairos) [reply .travis.yml]: And that's what I thought, but my test runs on travis were failing because none of the tables were getting created. I'll remove this line and cross my fingers that the issues I saw don't return

File: .travis.yml (hunk 1 above)
Comment (cadecairos) [reply .travis.yml]: I was having some Database issues so I wanted to pin the version we test with to the same one we use in production. Travis defaults to 9.2 but we run 9.6 in prod.

**Comment by Pomax** [comment]
Is this in an easily rebasable form, or will this need to (effectively) be redone?

**Comment by cadecairos** [comment]
I'll rebase and test it, then finish the last change request that @patjouk made in her review.

**Comment by cadecairos** [comment]
Rebased and updated. ready for a final review.
